homeassistant/components/zha/core/channels/__init__.py

```diff
@@ -5,404 +5,314 @@
+        return self._channels.zha_device.nwk
 
-    async def async_initialize(self, from_cache):
-        """Initialize listener."""
-        await self.get_attribute_value(
-            self._report_config[0].get("attr"), from_cache=from_cache
-        )
-        await super().async_initialize(from_cache)
+    @property
+    def manufacturer(self) -> Optional[str]:
+        """Return device manufacturer."""
+        return self._channels.zha_device.manufacturer
 
+    @property
+    def manufacturer_code(self) -> Optional[int]:
+        """Return device manufacturer."""
+        return self._channels.zha_device.manufacturer_code
 
-class ZDOChannel(LogMixin):
-    """Channel for ZDO events."""
+    @property
+    def model(self) -> Optional[str]:
+        """Return device model."""
+        return self._channels.zha_device.model
 
-    def __init__(self, cluster, device):
-        """Initialize ZDOChannel."""
-        self.name = CHANNEL_ZDO
-        self._cluster = cluster
-        self._zha_device = device
-        self._status = ChannelStatus.CREATED
-        self._unique_id = "{}:{}_ZDO".format(str(device.ieee), device.name)
-        self._cluster.add_listener(self)
+    @property
+    def relay_channels(self) -> Dict[str, zha_typing.EventRelayChannelType]:
+        """Return a dict of event relay channels."""
+        return self._relay_channels
 
     @property
     def unique_id(self):
         """Return the unique id for this channel."""
         return self._unique_id
 
-    @property
-    def cluster(self):
-        """Return the aigpy cluster for this channel."""
-        return self._cluster
-
-    @property
-    def status(self):
-        """Return the status of the channel."""
-        return self._status
+    @classmethod
+    def new(cls, channels: Channels, ep_id: int) -> "ChannelPool":
+        """Create new channels for an endpoint."""
+        pool = cls(channels, ep_id)
+        pool.add_all_channels()
+        pool.add_relay_channels()
+        zha_disc.PROBE.discover_entities(pool)
+        return pool
 
     @callback
-    def device_announce(self, zigpy_device):
-        """Device announce handler."""
-        pass
+    def add_all_channels(self) -> None:
+        """Create and add channels for all input clusters."""
+        for cluster_id, cluster in self.endpoint.in_clusters.items():
+            channel_class = zha_regs.ZIGBEE_CHANNEL_REGISTRY.get(
+                cluster_id, base.AttributeListeningChannel
+            )
+            # really ugly hack to deal with xiaomi using the door lock cluster
+            # incorrectly.
+            if (
+                hasattr(cluster, "ep_attribute")
+                and cluster.ep_attribute == "multistate_input"
+            ):
+                channel_class = base.AttributeListeningChannel
+            # end of ugly hack
+            channel = channel_class(cluster, self)
+            if channel.name == const.CHANNEL_POWER_CONFIGURATION:
```

Comment:
we gotta make sure we only take this once from devices and i would stick to only taking it from the ZHA or ZLL profile. Centralite for instance has their own profile with all of the clusters on it. There are others too.

Reply:
Yes, taking in only the 1st `PowerConfiguration` channel/cluster even if multiple are available. Have tests for that too.